packages/angular/cli/src/analytics/analytics-collector.ts

```diff
@@ -52,8 +52,7 @@ export class AnalyticsCollector {
 
     this.requestParameterStringified = querystring.stringify(requestParameters);
 
-    // Remove the `v` at the beginning.
-    const nodeVersion = process.version.substring(1);
+    const nodeVersion = parseVersion(process.version);
```

Comment:
```suggestion
    const nodeVersion = semver.clean(process.version);
```

Reply:
Thanks for pointing that out, I wasn't aware of `semver.clean()`. Though I think that still leaves in the user-defined pre-release? https://www.npmjs.com/package/semver indicates that `semver.clean('2.1.5-test') === '2.1.5-test'`, so there's still an unbounded string there we need to strip.

Reply:
Your `parseVersion` also keeps the prerelease string. 

Couple of ways you can remove the prerelease.

```ts
const version = '2.1.5-test';

// Using clean
let newVersion = semver.clean(version).split('-', 1)[0];

// Using parse
const [major, minor, patch] = semver.parse(version);
let newVersion = `${major}.${minor}.${patch}`
```

The latter can also be used to populate `UserCustomDimension.NodeMajorVersion`

Reply:
Ah whoops, I got so focused on the leading `v` I forgot about the prerelease :sweat_smile:. I like your `semver.parse()` approach, updated to use that.